### ISSUE TITLE
Burp 2 beta errors

### ISSUE THREAD

**Issue by Hipapheralkus** [augustd/burp-suite-software-version-checks#73]
Hi,
maybe it's not connected, but I get Java exceptions when trying out on the new Burp 2:
```
java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:901)
	at java.util.ArrayList$Itr.next(ArrayList.java:851)
	at com.codemagi.burp.PassiveScan.runPassiveScanChecks(PassiveScan.java:120)
	at burp.BurpExtender.processHttpMessage(BurpExtender.java:209)
	at burp.beh.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
java.lang.IndexOutOfBoundsException: No group 2
	at java.util.regex.Matcher.group(Matcher.java:538)
	at com.codemagi.burp.PassiveScan.runPassiveScanChecks(PassiveScan.java:126)
	at burp.BurpExtender.processHttpMessage(BurpExtender.java:209)
	at burp.beh.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)

```

**Comment by augustd** [comment]
@Hipapheralkus This type of error is usually due to a partial match or a false positive. If you know what URL/response caused the error I can try to track down which match rule is causing the error and "tune out" the false positive.

**Comment by augustd** [comment]
I think this is a duplicate of #72, which is now fixed by PR #83. Please let me know if you continue to get this error with the latest match rules.
